### PR TITLE
[Feature] Improve current flow detail

### LINKED ISSUE #29: [Feature] Improve current flow detail
## Feature Description

The flows of the current systems and suggested bond/guarantor approach are missing some details, e.g., 
- [x] ERC-20 fungible tokens
- [ ] prioritizing that delegation needs to happen before the snapshot proposal
- [x] partial slash capability
- [x] rewards for bonding are not shown
- [ ] integrating bonds into existing defi systems (e.g., Vortex, Risk Harbor)
- [ ] describe the UIs
- [x] user expertise level
- [x] number of actors represented by a depiction
- [ ] potential use of snapshot plugins
- [x] redemption flow from the bond

### PR DESCRIPTION
**Pull request by shahankhatch**
Closes #29 

The details that are left incomplete from the issue will need to be taken care of in a different flow:
- rewards and insurance mechanisms involve more actors and decision maker processes
- UIs should be described elsewhere
- motivating that delegation needs to occur prior to the snapshot proposal is low priority
- snapshot plugins should be explored in a separate issue

### CODE REVIEW

File: docs/flows/flows.puml [hunk 1]
```diff
@@ -70,33 +72,52 @@ autonumber "Bond:[00]"
 
 alice -> forum: create forum proposal
 alice <- guarantor: positive/negative feedback
-alice -> bondcontract: if (positive) create bond
+alice -> bondcontract: if (positive) create bond (specify reward mechanism)
 
-guarantor -> bondcontract: contribute
+guarantor -> bondcontract: contribute BIT tokens
+bondcontract <-- guarantor: ERC20 Fungible bond tokens (1:1 swap)
```
Comment (CjHare) [inline docs/flows/flows.puml:78]: Are we specifying here whether the fungible token is fungible within the bond contract or fungible within all bond contracts?
_(For secondary markets, I can see the value in keeping each bond contract as unique pools if fungible tokens)_

File: docs/flows/flows.puml [hunk 2]
```diff
@@ -70,33 +72,52 @@ autonumber "Bond:[00]"
...
-alice -> snapshot: submit proposal
+alice -> snapshot: if (positive) submit proposal
+
+tokenholders -> snapshot: show proposals
+tokenholders <-- snapshot: proposals
+tokenholders -> snapshot: vote on open proposals
+tokenholders <- snapshot: vote
 
 bitdaoadmin -> bitdaotreasury: if (pass) allow funds
 bitdaotreasury -> grantstreasury: transfer funds
```
Comment (CjHare) [inline docs/flows/flows.puml:90]: Funds transfer seem to fall into two family of approaches:
1) Give all the funds; when the `bitdaoadmin` allows funds, the passed amount is transferred to the receiving treasury.
2) Pull release; when the `bitdaoadmin` allows funds, a limit is set allowing the receiving treasury to pull funds up to that limit. 

The difference being the locus of control for triggering the transfers.

**Review by CjHare — COMMENTED** [review]
The release / slashing changes sound good 👍

File: docs/flows/flows.puml (hunk 1 above)
Comment (shahankhatch) [reply docs/flows/flows.puml:78]: We can scope token fungibility to within the bond contract; a swap can be used to be fungible within another bond contract. This view can be translated into fungibility within all bond contracts by swapping tokens for another derivative which is an index across all bond contracts.

File: docs/flows/flows.puml (hunk 2 above)
Comment (shahankhatch) [reply docs/flows/flows.puml:90]: Good points. This should be captured in some spec with tradeoffs and guidelines.

File: docs/flows/flows.puml (hunk 1 above)
Comment (CjHare) [reply docs/flows/flows.puml:78]: Sounds good 👍

**Pull request merged**